charts/datadog/templates/agent-cilium-network-policy.yaml

```
@@ -108,6 +115,52 @@ specs:
           - ports:
               - port: "10250"
                 protocol: TCP
+              - port: "6942"
+                protocol: "TCP"
+  - description: "Egress to Prometheus metrics"
+    endpointSelector:
+      matchLabels:
+        app: {{ template "datadog.fullname" . }}
+        {{- if .Values.agents.podLabels }}
+        {{ toYaml .Values.agents.podLabels | indent 8 }}
+        {{- end }}
+    egress:
+      - toEntities:
+          - host
+        toPorts:
+          - ports:
+              - port: "9090"
+                protocol: "TCP"
+  - description: "Egress to cluster agent"
+    endpointSelector:
```

Cluster agent specific parts should be generated only if the cluster agent is enabled.
```suggestion
{{- if eq (include "cluster-agent-enabled" .) "true" }}
  - description: "Egress to cluster agent"
    endpointSelector:
```